packages/echo-core/src/__tests__/actions/coreActions/extensions.test.ts

```diff
@@ -0,0 +1,151 @@
+            // given
+            const extensionsToRegister: ExtensionRegistration[] = [
+                {
+                    ...extension
+                },
+                {
+                    ...extension,
+                    key: 'key-1'
+                },
+                {
+                    ...extension,
+                    key: 'key-2'
+                },
+                {
+                    ...extension,
+                    key: 'key-0',
+                    extends: 'TheBestEchopediaWebComponent'
+                }
+            ];
+
+            // when
+            const unregisterFunctions = registerMultipleExtensions(extensionsToRegister);
+
+            // then
+            expect(getExtensions()).toStrictEqual({
+                MyFavEchopediaWebComponent: [
+                    {
+                        key: 'unique-key',
+                        extends: 'MyFavEchopediaWebComponent',
+                        component: mockComponent
+                    },
+                    {
+                        key: 'key-1',
+                        extends: 'MyFavEchopediaWebComponent',
+                        component: mockComponent
+                    },
+                    {
+                        key: 'key-2',
+                        extends: 'MyFavEchopediaWebComponent',
+                        component: mockComponent
+                    }
+                ],
+                TheBestEchopediaWebComponent: [
+                    {
+                        key: 'key-0',
+                        extends: 'TheBestEchopediaWebComponent',
+                        component: mockComponent
+                    }
+                ]
+            });
+
+            // when
+            unregisterFunctions[0]();
+            unregisterFunctions[2]();
+
+            // then
+            expect(getExtensions()).toStrictEqual({
+                MyFavEchopediaWebComponent: [
+                    {
+                        key: 'key-1',
+                        extends: 'MyFavEchopediaWebComponent',
+                        component: mockComponent
+                    }
+                ],
+                TheBestEchopediaWebComponent: [
+                    {
+                        key: 'key-0',
+                        extends: 'TheBestEchopediaWebComponent',
+                        component: mockComponent
+                    }
+                ]
+            });
+
+            // when
+            unregisterFunctions[1]();
+            unregisterFunctions[3]();
+
+            // then
+            expect(getExtensions()).toStrictEqual({
+                MyFavEchopediaWebComponent: [],
+                TheBestEchopediaWebComponent: []
+            });
+        });
+    });
+});
+
+const mockComponent = () => {
+    return null;
+};
+
+const extension: ExtensionRegistration = {
+    key: 'unique-key',
+    extends: 'MyFavEchopediaWebComponent',
+    component: mockComponent
+};
+
+function getExtensions() {
```

Comment:
Missing return type on function.eslint

Reply:
Hmm, I don't get this error. Also this is just a private helper function in one smaller test file. I don't think types here add much value.